### PR TITLE
Bm delete empty global configs

### PR DESCRIPTION
**Pull request by bamandel**
If you delete all fields and try to save an empty global configuration for Polaris/BlackDuck/Email/Hipchat, you will instead hit the delete endpoint and delete the old configuration.

The FieldsPanel also had a bug and has been fixed to allow this deletion.

### CODE REVIEW

**Comment by bamandel** [comment]
> Hopefully the duplicate code will be cut down when this is fully dynamic.

When we go dynamic, most of these files will be deleted. The global dynamic page has this feature built into it as well.

File: src/main/js/channels/EmailConfiguration.js [hunk 1]
```diff
@@ -205,7 +210,13 @@ class EmailConfiguration extends React.Component {
         evt.preventDefault();
         evt.stopPropagation();
         const fieldModel = this.state.currentEmailConfig;
-        this.props.updateEmailConfig(fieldModel);
+        let emptyModel = !FieldModelUtilities.hasAnyValuesExcludingId(fieldModel);
+        let id = FieldModelUtilities.getFieldModelId(fieldModel);
```
Comment (jamesrichard91) [inline src/main/js/channels/EmailConfiguration.js]: Change emptyModel and id to `const` variables

File: src/main/js/channels/HipChatConfiguration.js [hunk 2]
```diff
@@ -65,7 +70,13 @@ class HipChatConfiguration extends React.Component {
         event.preventDefault();
         event.stopPropagation();
         const fieldModel = this.state.currentHipChatConfig;
-        this.props.updateConfig(fieldModel);
+        let emptyModel = !FieldModelUtilities.hasAnyValuesExcludingId(fieldModel);
+        let id = FieldModelUtilities.getFieldModelId(fieldModel);
```
Comment (jamesrichard91) [inline src/main/js/channels/HipChatConfiguration.js]: Change emptyModel and id to const variables

File: src/main/js/dynamic/GlobalConfiguration.js [hunk 3]
```diff
@@ -31,28 +32,49 @@ class GlobalConfiguration extends React.Component {
...
 
-    handleSubmit(evt) {
-        evt.preventDefault();
+    handleSubmit(event) {
+        event.preventDefault();
+        event.stopPropagation();
         const fieldModel = this.state.currentConfig;
-        this.props.updateConfig(fieldModel);
+        let emptyModel = !FieldModelUtilities.hasAnyValuesExcludingId(fieldModel);
+        let id = FieldModelUtilities.getFieldModelId(fieldModel);
```
Comment (jamesrichard91) [inline src/main/js/dynamic/GlobalConfiguration.js]: Same

File: src/main/js/providers/BlackDuckConfiguration.js [hunk 4]
```diff
@@ -80,10 +85,17 @@ class BlackDuckConfiguration
...
 
-    handleSubmit(evt) {
-        evt.preventDefault();
+    handleSubmit(event) {
+        event.preventDefault();
+        event.stopPropagation();
         const fieldModel = this.state.currentConfig;
-        this.props.updateConfig(fieldModel);
+        let emptyModel = !FieldModelUtilities.hasAnyValuesExcludingId(fieldModel);
+        let id = FieldModelUtilities.getFieldModelId(fieldModel);
```
Comment (jamesrichard91) [inline src/main/js/providers/BlackDuckConfiguration.js]: Same

**Pull request merged**